### PR TITLE
[SYCL][ESIMD] Prepare LLVM IR for OS spirv translator

### PR DESCRIPTION
**Pull request by NikitaRudenkoIntel**

### CODE REVIEW

**Comment by NikitaRudenkoIntel** [comment]
The Khronos translator does not have source language "6". In vector compute patches we use other approach to decide on how to translate VC-specific things

**Comment by kbobrovs** [comment]
@AGindinson - please re-approve

File: clang/lib/Driver/ToolChains/Clang.cpp [hunk 1]
```diff
@@ -7628,6 +7628,8 @@ void SPIRVTranslator::ConstructJob(Compilation &C, const JobAction &JA,
   if (getToolChain().getTriple().isSYCLDeviceEnvironment()) {
     TranslatorArgs.push_back("-spirv-max-version=1.1");
     std::string ExtArg("-spirv-ext=+all");
+    if (C.getArgs().hasArg(options::OPT_fsycl_esimd))
+      TranslatorArgs.push_back("-spirv-allow-unknown-intrinsics");
```
Comment (AGindinson) [inline clang/lib/Driver/ToolChains/Clang.cpp:7632]: Could you please add a quick test be to `clang/test/Driver/sycl-offload.c`? Something like:
```
/// RUN: %clangxx %s -fsycl -fsycl-esimd -### 2>&1 | FileCheck %s --check-prefix=CHK-FSYCL-ESIMD
/// CHK-FSYCL-ESIMD: llvm-spirv {{.*}} "-spirv-allow-unknown-intrinsics"
```
Sorry for missing out on this ealier.

File: clang/lib/Driver/ToolChains/SYCL.cpp [hunk 2]
```diff
@@ -42,6 +42,8 @@ const char *SYCL::Linker::constructLLVMSpirvCommand(Compilation &C,
   } else {
     CmdArgs.push_back("-spirv-max-version=1.1");
     CmdArgs.push_back("-spirv-ext=+all");
+    if (C.getArgs().hasArg(options::OPT_fsycl_esimd))
```
Comment (MrSidims) [inline clang/lib/Driver/ToolChains/SYCL.cpp:45]: Not in a scope of this patch, but I think we shall add another option to enable translation of unknown to the translator intrinsic, for example for users: https://github.com/intel/llvm/issues/2122

File: clang/test/Driver/sycl-offload.c [hunk 3]
```diff
@@ -593,6 +593,10 @@
 // CHECK-LINK-SYCL-DEBUG: "{{.*}}link{{(.exe)?}}"
 // CHECK-LINK-SYCL-DEBUG: "-defaultlib:sycld.lib"
 
+/// Check "-spirv-allow-unknown-intrinsics" option is emitted for llvm-spirv tool for esimd mode
+// RUN: %clangxx %s -fsycl -fsycl-explicit-simd -### 2>&1 | FileCheck %s --check-prefix=CHK-FSYCL-ESIMD
+// CHK-FSYCL-ESIMD: llvm-spirv {{.*}} "-spirv-allow-unknown-intrinsics"
```
Comment (kbobrovs) [inline clang/test/Driver/sycl-offload.c:598]:
```suggestion
// CHK-FSYCL-ESIMD:  {{.*}}llvm-foreach{{.*}} "--" {{.*}}llvm-spirv{{.*}} "-spirv-allow-unknown-intrinsics" {{.*}}
```

**Comment by kbobrovs** [comment]
@bader, CUDA buildbot failed with

CMake Error at /usr/local/share/cmake-3.15/Modules/FindCUDA.cmake:700 (message):
  Specify CUDA_TOOLKIT_ROOT_DIR
Call Stack (most recent call first):
  /localdisk2/sycl_ci/buildbot/worker/Lit_With_Cuda/llvm.src/sycl/plugins/cuda/CMakeLists.txt:7 (find_package)

This does not seem related to this patch. Since it is approved and all other checks passed - do you think it can be merged?

**Pull request merged**